universal-application-tool-0.0.1/app/services/program/ProgramService.java

```diff
@@ -0,0 +1,17 @@
+package services.program;
+
+import com.google.common.collect.ImmutableList;
+
+/** Operations you can perform on Programs. */
+public interface ProgramService {
+
+    /** List all programs. */
+    public ImmutableList<ProgramDefinition> listProgramDefinitions();
+
+    /** Get the definition for a given program. */
+    public ProgramDefinition getProgramDefinition(String id);
+
+    /** Create a new program. */
+    public ProgramDefinition createProgram(String name, String description);
```

Comment:
See comment [here](https://github.com/seattle-uat/universal-application-tool/pull/115/files#r574175484) on the possibility of surfacing validation errors here.

Reply:
Interesting! Are we preferring to return validation errors rather than throw? Or a mix of both? Just trying to wrap my head around these methods

Kendrick and I were wondering what kind of validation errors initial create would give, if it's just setting up a name and description

Reply:
We don't have to decide this now (forgot to specify this is a non-blocking discussion), but I'm trying to think about where the right place to surface validation errors is. For create program we'll probably want to validate that at the very least they include a name for the program, likely more stuff down the line (department, etc).

Reply:
That makes sense to me, thanks for clarifying! I suppose this will either need to throw if validation fails and something upstream can format the errors nicely, or it should return a validation error class as you suggested

Reply:
If it throws I'm not sure how it would be able to return multiple errors if there are multiple validation problems.

Reply:
Good point - I'm used to having validators that just throw on the first issue they see. But it would be a much nicer experience to see all your issues at once instead